### PR TITLE
Part 1 of Server API Docs: docs(server): cli commands docs

### PR DESCRIPTION
**Pull request by mtomcal**
- CLI Commands API Doc
- Stubbed out other Server API Doc files

### CODE REVIEW

File: docs/api/API.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
...
+**Holocron Modules** or "Modules" are self contained Web Experiences that consist of [React Components](https://reactjs.org/docs/components-and-props.html), [Redux](https://redux.js.org/) Reducers and Actions as well as [Routes](https://github.com/americanexpress/one-app-router). In practice, Modules are developed, bundled, and operate in isolation to one another. The One App Server uses a [Module Map](../../README.md#building-and-deploying-a-holocron-module-map) containing URLs to Module bundles (e.g. `my-module.browser.js`) to load and serve bundles upon request. When the Server receives an HTTP request, it renders one or more Modules on the Server. Similar to React Components, Modules are composable (e.g. Modules may load other Modules). The first or entrypoint Module is called the "Root Module". The Root Module loads other "Child Modules". Overall, this development pattern in One App may be characterized as the [Micro Front End](https://martinfowler.com/articles/micro-frontends.html) pattern.
```
Comment (mtomcal) [inline docs/api/API.md]: We just removed extra newlines.

Comment (mtomcal) [reply docs/api/API.md]: And I added more links to libraries.

File: docs/api/server/CLI-Commands.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+## Development Commands
+
+> 💬 These commands only work in Development
+
+| Flag | Description |
+|------|-------------|
+| `serve-module` | Accepts a path to a [Holocron Module] and adds it to the local development CDN. *See [`serve-module` Arguments](#serve-module-arguments).*  |
+| `drop-module` | Accepts a [Holocron Module] name and removes the Module from the local development CDN. *See [`drop-module` Arguments](#drop-module-arguments).* |
+| `set-middleware` | Accepts a Node JS file that defines custom [Express Middleware](http://expressjs.com/en/guide/writing-middleware.html#writing-middleware-for-use-in-express-apps) for the development proxy. *See [`set-middleware` Arguments](#set-middleware-arguments).* |
+| `set-dev-endpoints`| Accepts a Node file containing development endpoints for the development proxy. *See [`set-dev-endpoints` Arguments](#set-dev-endpoints-arguments).* |
```
Comment (jsmartfo) [inline docs/api/server/CLI-Commands.md]: > Accepts a Node file containing development endpoints for the development proxy

Can we elaborate on the development proxy a little bit more pls ... just a little detail on what it does. 

Somewhere in the md: maybe in a notes section or within some parenthesis when we first mention it

Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: > Can we elaborate on the development proxy a little bit more pls ... just a little detail on what it does.
This CLI Commands file is purely a reference file and I was cautioned from going into detail on those libraries and leave it to the "Development-Tools.md" effort.

File: docs/api/server/CLI-Commands.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  * [`test`](#test-commands)
+  * [`test:unit`](#test-commands)
+  * [`test:lint`](#test-commands)
+  * [`pretest:integration`](#test-commands)
+  * [`test:integration`](#test-commands)
+  * [`test:danger`](#test-commands)
+  * [`test:git-history`](#test-commands)
+  * [`test:lockfile`](#test-commands)
+  * [`prestart:prod-sample`](#test-commands)
+  * [`start:prod-sample`](#test-commands)
```
Comment (10xLaCroixDrinker) [inline docs/api/server/CLI-Commands.md]: do these belong under "Start commands"?

File: docs/api/server/CLI-Commands.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+|---------------|------------------------------------------------------------------------------------------------|
+| `build` | Using the [`concurrently`] package, runs `build:server` and `build:bundle`. |
+| `build:server` | Executes [Babel] to transpile ES6 Server source files |
+| `build:bundle` | Executes [One App Bundler] to build Browser assets |
+| `build:artifacts:statics` | Runs [Static Assets Script](../../../scripts/build-static-assets-artifact.js) for [Integration Tests] |
+| `build:prod-sample` | Runs `build:sample-modules` and runs the [Build One App Docker Setup Script](../../../scripts/build-one-app-docker-setup.js) for [Integration Tests] |
+| `build:sample-modules` | Builds the [Holocron Modules] in [`prod-sample/sample-modules`] folder for [Integration Tests] |
+
+```
+npm run build:sample-modules -- --archive-built-artifacts --bundle-statics-hostname=https://cdn.example.com
```
Comment (10xLaCroixDrinker) [inline docs/api/server/CLI-Commands.md]: can you use the [docopt](http://docopt.org/) format throughout?

File: docs/api/server/CLI-Commands.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+| `test:git-history` | Validates the Git commit message format using the [`commitlint`](https://github.com/conventional-changelog/commitlint) tool. |
+| `test:lockfile` | Validates that the `package-lock.json` file contains valid NPM registry URLs using the [`lockfile-lint`](https://github.com/lirantal/lockfile-lint) tool. |
+| `prestart:prod-sample` | Runs [`build:sample-modules`](#build-commands) before `start:prod-sample`. |
+| `start:prod-sample` | Runs a [Docker Compose](https://docs.docker.com/compose/compose-file/) file to start up the [Integration Test Suite] containers. |
+
+## Development Commands
+
+> 💬 These commands only work in Development
+
+| Flag | Description |
```
Comment (10xLaCroixDrinker) [inline docs/api/server/CLI-Commands.md]: these arent flags

File: docs/api/server/CLI-Commands.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+|------|-------------|
+| `serve-module` | Accepts a path to a [Holocron Module] and adds it to the local development CDN. *See [`serve-module` Arguments](#serve-module-arguments).*  |
+| `drop-module` | Accepts a [Holocron Module] name and removes the Module from the local development CDN. *See [`drop-module` Arguments](#drop-module-arguments).* |
+| `set-middleware` | Accepts a Node JS file that defines custom [Express Middleware](http://expressjs.com/en/guide/writing-middleware.html#writing-middleware-for-use-in-express-apps) for the development proxy. *See [`set-middleware` Arguments](#set-middleware-arguments).* |
```
Comment (10xLaCroixDrinker) [inline docs/api/server/CLI-Commands.md]: > Accepts a Node JS file

The argument is a path to a file, not a file. I don't think its necessary to specify "Node"

File: docs/api/server/CLI-Commands.md (hunk 3 above)
Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: Depends on how you see this. Im open to moving `start:prod-sample` to Start Commands.

File: docs/api/server/CLI-Commands.md (hunk 5 above)
Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: Arguments?

File: docs/api/server/CLI-Commands.md (hunk 3 above)
Comment (10xLaCroixDrinker) [reply docs/api/server/CLI-Commands.md]: To me, it makes more sense under start commands

File: docs/api/server/CLI-Commands.md (hunk 4 above)
Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: Fixed in a88378aafc426e868db005159f4e769323b6a93d

File: docs/api/server/CLI-Commands.md (hunk 5 above)
Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: Fixed in a88378aafc426e868db005159f4e769323b6a93d

File: docs/api/server/CLI-Commands.md (hunk 6 above)
Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: Fixed in a88378aafc426e868db005159f4e769323b6a93d

File: docs/api/server/CLI-Commands.md (hunk 3 above)
Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: Fixed in a88378aafc426e868db005159f4e769323b6a93d

File: docs/api/server/CLI-Commands.md (hunk 2 above)
Comment (stephaniecoates) [reply docs/api/server/CLI-Commands.md]: how about hyperlinking to the github repos (or the `Development-Tools.md`, whichever gives better context) when you first declare both `local development cdn` and `development proxy`?

File: docs/api/server/CLI-Commands.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+  * [`test:danger`](#test-commands)
+  * [`test:git-history`](#test-commands)
+  * [`test:lockfile`](#test-commands)
+* Development Commands
+  * [`serve-module`](#test-commands)
+  * [`drop-module`](#test-commands)
+  * [`set-middleware`](#test-commands)
+  * [`set-dev-endpoints`](#test-commands)
+
+## Install Commands
```
Comment (anescobar1991) [inline docs/api/server/CLI-Commands.md]: Most of these are not really part of the One App API, they should be documented in the CONTRIBUTING.md file instead IMO.

Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: Talked offline and agreed to reordering topics based on User centric versus Contributor centric commands.

File: docs/api/server/CLI-Commands.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+
+| Command       | Description                                                                                    |
+|---------------|------------------------------------------------------------------------------------------------|
+| `preinstall`  | Runs `check-engines` package to ensure a compatible Node version is used before `npm install`. |
+| `postinstall` | Runs a [`build`](#build-commands) after `npm install`.                                         |
+
+## Development Commands
+
+> 💬 These commands only work in Development
```
Comment (anescobar1991) [inline docs/api/server/CLI-Commands.md]: Technically these commands all are only for development right?

Comment (mtomcal) [reply docs/api/server/CLI-Commands.md]: What do you mean? I have this callout under Development Commands already in there:

> 💬 These commands only work in Development

Comment (JAdshead) [reply docs/api/server/CLI-Commands.md]: what about rephrasing to "These commands are for Development only"

**Pull request merged**